src/view/ScanItemView.java

```diff
@@ -0,0 +1,162 @@
+package view;
+
+import interface_adapter.Dashboard.DashboardViewModel;
+import interface_adapter.ScanItem.ScanItemController;
+import interface_adapter.ScanItem.ScanItemState;
+import interface_adapter.ScanItem.ScanItemViewModel;
+
+import javax.swing.*;
+import javax.swing.event.DocumentEvent;
+import javax.swing.event.DocumentListener;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+import java.io.IOException;
+
+public class ScanItemView extends JPanel implements ActionListener, PropertyChangeListener {
+    public final String viewName = "scan item";
+    private final ScanItemViewModel scanItemViewModel;
+    private final ScanItemController scanItemController;
+    private final DashboardViewModel dashboardViewModel;
+    JPanel scanRightPanel;
+    private JRadioButton scanFileRadioButton;
+    private JRadioButton scanURLRadioButton;
+    private JButton selectFileButton;
+    private JTextField typeURLTextField;
+    private JPanel main;
+    private JButton confirmButton;
+    private JPanel scanPanel;
+    private JButton backButton;
+
+    public ScanItemView(ScanItemViewModel scanItemViewModel,
```

Comment:
Although this is the constructor method, it might still be helpful to include a Javadoc here